haskellPackages: `justStaticExecutables` doesn't work together with `enableExecutableProfiling`

### Describe the bug
When having both of them enabled, ghc fails with:
```
┃        > bla/Module.hs:29:1: error:
┃        >     Could not find module ‘The.Module`
┃        >     Perhaps you haven't installed the profiling libraries for package ‘the-package’?
┃        >     Use -v (or `:set -v` in ghci) to see a list of the files searched for.
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. enable executable profiling
2. enable static executables

### Expected behavior

both of them should go together 


### Notify maintainers

@maralorn @sternenseemann 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/haskell-modules/lib/compose.nix
+++ b/pkgs/development/haskell-modules/lib/compose.nix
@@ -292,7 +292,7 @@ rec {
    */
  justStaticExecutables = overrideCabal (drv: {
     enableSharedExecutables = false;
-    enableLibraryProfiling = false;
+    enableLibraryProfiling = drv.enableExecutableProfiling or false;
     isLibrary = false;
     doHaddock = false;
     postFixup = drv.postFixup or "" + ''

